### DISCUSSION TITLE
Requests? "Uturns" Yes/No and next turn street name to be shown earlier?

### DISCUSSION THREAD

**Discussion by mapgull** [organicmaps/organicmaps#4437] [Ideas]
Hi. just a user here. I really like Organic maps. I've been using it lately fairly intensely where I'm not exactly sure how to get to a place while surrounded by hi-speed traffic that DOES know where it's going ...Organic maps is nice and simple and I generally don't have to (or want to) touch it in the midst of it all...

  So, preamble aside, (using version 22121812 on Android11 on a 15x7mm screen) I would like to see an option for No "U-turns". It's a little annoying when OM wants me to rotate on a freeway.. even though  u-turns have their place...  Come to think, I can't actually think of a time when I needed to be TOLD a uturn was wanted on any gps or app.I think it's pretty plain to most drivers when they miss it-that sinking feeling...

Another request: When a turn is coming, it'd be a help to see the name of that street ahead of time. It doesn't seem to show up until seconds before the turn and often (long) before that time, I have to have committed myself to a lane, or highway exit. It can be a help to know ahead of time especially where the wrong turn is very near the correct turn...(yes, my eyes are fairly good and even a distant blur can be interpreted when there's fore-knowledge)  I do notice that it starts out ok when I "start" the route, showing the target turn right away but, in the midst of it , I was waiting and waiting for a confirmation of the name....

font on the screen is ok in default but I think in particular the street name could be a little bigger when it does finally appear.

**Comment by biodranik** [comment]
Can you please clarify about U-turns? Did OM suggest wrong ones? If it's OpenStreetMap.org data issue, then it may be better to fix it there.

Regarding the street name, doesn't it appear together with a turn? Do you suggest showing turns earlier?

**Comment by mapgull** [comment]
Hi,biodranik...
  per U-turns, I'm  suggesting a button to _enable_ or _disable_ U-turns, on the list  of Setttings/Route options which presently has these with choices: Toll Roads,Unpaved roads,Ferry crossings,Motorways...specifically to your comment, a ""wrong" u-turn would be _any_ u-turn on a "motorway". What would be even more  helpful, would be re-routing through the next exit via a cloverleaf or what else was _practically_ available at 100+kmh. But I don't remember using any application yet that will do this when U-turns are enabled. It'd be nice if a U-turns setting knew the difference between sidestreets and "motorways" but just turning u-turns off is the one I 'm hoping for now.

  per streetnames of turn, it's just something I noticed "in action" on city streets as my description above might suggest.  The street names all exist on the Open Street map. But when I looked for it-it wasn't there until it was too late to be useful, (unless it happened a traffic light was red etc..and my  turn was very close.... )  Voice directions..well- they are always problematic on any device or app I've used and  I look to the route-line as a guide. A street name does help avoid some confusions...
  In my opinion it'd be most useful if it appeared as soon as the "new direction" does.   (and Yes, p-I'm describing navigation at major cross streets not remote country lanes if that's what you mean) 
Perhaps the lag is in my hardware

**Reply by biodranik** [reply]
1. I was interested if you saw some wrong U-turns in reality and can tell some examples where it was suggested by OM (it should be fixed properly in osm.org). The idea to disable _valid_ u-turns sounds interesting, but I don't remember a lot of u-turns in route suggestions. Do you have any examples? Do other nav apps have this option?
2. Isn't the street name appear now together with a new direction? Does it appear later?

CC @AntonM030481

**Comment by mapgull** [comment]
per 

>"I was interested if you saw some wrong U-turns in reality and can tell some examples where it was suggested by OM"
 
I certainly did. As I hinted but perhaps not plain enough, any Uturn on a 4 or more lane highway is wrong and hereabouts anyways, illegal even where it is physically possible without barriers, ditches, and any traffic on the other side of that.
  I missed an exit from the 4-or-so lane hwy and what I saw and (faintly heard) was the route line ending shortly ahead of my present position and nonsense about uturns. Fortunately, I am used to several varieties of garmins,magellans, TomToms and android gps apps and I know when to ignore them.  I certainly didn't  take notes,screenshots or tweak it  at the time, seeing as it was pouring rain and 4 or more merging lanes into a tunnel....likely the tunnel affected "recovery" but there was time before that and plenty once out of the tunnel. This phone has a very quick gps so I'm allowing minimal excuse on that event.
   It's an aside  because the u-turn's the issue, but at the time, I was sort of hoping for a reroute as almost any old Garmin will do when it gives up on the u-turn idea but OM was notably slower to figure it out . Finally I was offered a new route .
You can follow along from here,
49.05245343637279,-123.05742445354463
aiming for  here, ideally
49.13153514147746,-122.56969156627656

notice that by wringing its hands about a Uturn, it missed the obvious exit BEFORE the tunnel...and didn't recover its wits until roughly
49.15963582358228,-123.08673563365937
